[4851] Issues with plugin management

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4851
https://github.com/craftercms/craftercms/issues/4719

aachavarria committed:
Fixing issues related with the rootPath,
Adding setTimeout to initial pathNavigatorTreeBackgroundRefresh

aachavarria committed:
adding comment

aachavarria committed:
removing console.log

aachavarria committed:
deleting comment

aachavarria committed:
adding progress bar to plugins being installed;
updating minimizedbar styles;

aachavarria committed:
removing comment

rart commented on ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx:
```diff
@@ -495,6 +508,8 @@ export default function PathNavigatorTree(props: PathNavigatorTreeProps) {
     if (option === 'refresh') {
       state.expanded.forEach((path) => fetchingPathsRef.current.push(path));
+      console.log(state);
+      console.log(fetchingPathsRef.current.join(', '));
```
x

rart commented on ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx:
```diff
@@ -174,6 +175,17 @@ export default function PathNavigatorTree(props: PathNavigatorTreeProps) {
+  useMount(() => {
+    if (state) {
+      // If the state is being restoring from redux, fetchingPathsRef is being processed.
+      // Added a timeout to trigger the process again with a new fetchingPathsRef
```
Ping me to review. Didn't quite understand from the comment.
```suggestion
      // If the state is being restored from redux, fetchingPathsRef is being processed.
      // Added a timeout to trigger the process again with a new fetchingPathsRef
```

rart commented on ui/app/src/components/SystemStatus/MinimizedBar.tsx:
```diff
@@ -49,8 +50,10 @@ export function MinimizedBar(props: MinimizedBarProps) {
   const classes = useStyles({});
```
```suggestion
  const classes = useStyles();
```

rart committed:
Grammar

rart reviewed (changes requested):
1. Assess moving fetchingByPath to redux state
2. Check/remove compilation **warnings**

aachavarria committed:
Refactor on PathNavigatorTree.tsx to move component state to redux

aachavarria committed:
Adding events propagation;
Fetching RootItem;

aachavarria committed:
using memo

aachavarria committed:
Cleaning warnings

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4851

aachavarria committed:
removing warnings

aachavarria committed:
adding rootNode validation

aachavarria committed:
Fixing issue on LegacyFormDialog.tsx to close in progress;
Fixing issues on PathNavigatorTree.tsx related to nodes references
Fixing issue on DeleteDialog.tsx no sending selected items to itemsDeleted event

rart commented on ui/app/src/components/Dialogs/LegacyFormDialog.tsx:
```diff
@@ -316,6 +316,9 @@ export default function LegacyFormDialog(props: LegacyFormDialogProps) {
 
   const onClose = () => {
+    if (inProgress) {
+      props.onClose();
```
```suggestion
      props?.onClose();
```

rart commented on ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx:
```diff
@@ -504,8 +480,8 @@ export default function PathNavigatorTree(props: PathNavigatorTreeProps) {
-    nodesByPathRef.current[path].children = [{ id: 'loading' }];
-    setRootNode({ ...nodesByPathRef.current[rootPath] });
+    // nodesByPathRef.current[path].children = [{ id: 'loading' }];
+    // setRootNode({ ...nodesByPathRef.current[rootPath] });
```
x
```suggestion
```

rart commented on ui/app/src/state/epics/pathNavigatorTree.ts:
```diff
@@ -87,10 +85,7 @@ export default [
-          map((item) => pathNavigatorTreeFetchRootItemComplete({ id, item })),
-          catchAjaxError((error) => pathNavigatorTreeFetchRootItemFailed({ error, id }))
-        );
+        return NEVER;
```
`filter(), switchMap()` instead of returning NEVER

aachavarria committed:
adding filter and switchMap to pathNavigatorTreeInit
